client/gutenberg/editor/calypsoify-iframe.jsx

```diff
@@ -29,16 +51,87 @@ const getIframeUrl = ( siteAdminUrl, postId, postType ) => {
 };
 
 class CalypsoifyIframe extends Component {
+	state = {
+		isMediaModalVisible: false,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.iframe = null;
+	}
+
+	componentDidMount() {
+		window.addEventListener( 'message', this.onMessage, false );
+	}
+
+	componentWillUnmount() {
+		window.removeEventListener( 'message', this.onMessage, false );
+	}
+
+	setIframeRef = element => {
+		this.iframe = element;
+	};
+
+	onMessage = ( { data } ) => {
+		if ( typeof data !== 'string' || data[ 0 ] !== '{' ) {
+			return;
+		}
```

Comment:
are we trying to guess if the data is JSONable? if so, we can use `try/catch`

```js
let data;
try {
	data = JSON.parse( data );
} catch ( e ) {
	return;
}
```

or use a common helper I like

```js
const parseJson = s => {
	try {
		return JSON.parse( s );
	} catch ( e ) {
		return;
	}
}

onMesage = ( { data } ) => {
	const data = parseJson( data );
	if ( undefined === data ) {
		return;
	}
}
```